docs/reference/snapshot-restore/register-repository.asciidoc

```diff
@@ -162,6 +162,9 @@ unit, for example: `1GB`, `10MB`, `5KB`, `500B`. Defaults to `null` (unlimited c
 `max_restore_bytes_per_sec`:: Throttles per node restore rate. Defaults to unlimited. Note that restores are also throttled through <<recovery,recovery settings>>.
 `max_snapshot_bytes_per_sec`:: Throttles per node snapshot rate. Defaults to `40mb` per second.
 `readonly`:: Makes repository read-only.  Defaults to `false`.
+`max_number_of_snapshots`:: Limits the maximum number of snapshots that the repository may contain. Defaults to `500`. Note that snapshot repositories do not
+scale indefinitely in size and might lead to master node performance and stability issues if grown past a certain size. Increasing this setting
+is generally not advisable and should be weighted carefully against alternatives such as deleting older snapshots or creating multiple repositories.
```

Comment:
```suggestion
Instead you should delete older snapshots or use multiple repositories.
```

Is creating multiple repositories really a solution?

Reply:
I guess technically yes, not a great one but you could run two (or more) SLM policies that take turns creating snapshots I suppose. That way you could have a much longer retention time at high resolution. 
Also, if you just want certain snapshots to live for a long time maybe it's a solution.
... certainly all of these aren't super ergonomically but I guess for some use-cases these are valid?